moderntribe/square-one
# Travisci test

This PR introduces TravisCI to square-one (currently passing integration tests!)

![image](https://user-images.githubusercontent.com/1066195/38153057-017fdb1c-3428-11e8-8e5f-cb59106ace1a.png)

### Important Notes
- This builds our entire docker environment on TravisCI!
- dev/docker/start.sh has had a number a changes made to it that can affect our local environment.
- start.sh now has automated, token suggestions using the Project ID and the git repo name, e.g. 'square1%20-%20square-one'
- In order to automate the composer install on TravisCI, we are using a Personal Access Token provided by Matt from the mt-travisci user. If this token ever changes, the build will fail. This token is encrypted in .travis.yml using [this](https://docs.travis-ci.com/user/environment-variables/#Encrypting-environment-variables) method.

### How to test
- TravisCI is already configured to run on new PR's, so it will automatically run and can be viewed [here](https://travis-ci.com/moderntribe/square-one/).
- As noted, test this on local as well due to the dev/docker/start.sh changes to make sure that it still asks for a github token and composer continues to work as it did before.
- **Windows:** I HAVE NOT yet tested this command fully works on Windows ``basename `git rev-parse --show-toplevel`` relying on @MZAWeb to validate this properly.

I have made some assumptions here:
- Everyone has git installed
- basename works on Windows/Git bash (I did test this)
